pkg/generate/ack/output.go

```diff
@@ -145,6 +145,23 @@ func CreateGenerationMetadata(
 	return nil
 }
 
+// LoadGenerationMetadata read the generation metadata for a given api version and
+// apis path.
+func LoadGenerationMetadata(apisPath, apiVersion string) (*GenerationMetadata, error) {
+	filePath := filepath.Join(apisPath, apiVersion, outputFileName)
+	b, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var generationMetadata GenerationMetadata
+	err = yaml.Unmarshal(b, &generationMetadata)
+	if err != nil {
+		return nil, err
+	}
+	return &generationMetadata, nil
+}
+
```

Comment:
@RedbackThomson If you can implement a similar function in your PR, i'm taking it :)

Reply:
These have been moved to `pkg/metadata`